dashboard/app/controllers/regional_partners_controller.rb

```diff
@@ -80,6 +80,21 @@ def search_program_manager
     render :show
   end
 
+  # POST /regional_partners/:id/add_mapping
+  def add_mapping
+    region = params[:region]
+    state = region if region.present? && region.in?(STATE_ABBR_WITH_DC_HASH.keys.map(&:to_s))
+    zip_code = region if region.present? && RegexpUtils.us_zip_code?(region)
+    @regional_partner.mappings.find_or_create_by(state: state, zip_code: zip_code)
```

Comment:
Do we want to parse the format instead of being more specific (e.g. `params[:region][:zip_code]`)? What if it's neither format? What if we add more mapping fields later?

Reply:
Similarly we might want to display these more clearly (e.g. in columns for state/zip) in the table. Though clearly it's easy to distinguish states and zips if these are all we end up using.

Reply:
Are we still going to support school district mapping, or is that going away?

Reply:
My current thinking is that the state and zip_code mappings should be stored in separate attributes in the model so that the application / regional partner matching logic can easily query it, but the user would find it easiest to enter any type of mapping (state or zip code) in a single field, perhaps enhanced with selectize.  This approach I think would continue to work if other types of mappings were added in the future (like District).

I am adding more error handling to the controller for scenarios where the input is not a valid state or zip code, but selectize should also reduce the risk of that happening.

Reply:
I like that user experience, however I still think it should be split at that point and not sent to the server for parsing. The selectize will have several sets of data (zip and state at least, school districts and maybe more later), and it knows the source of each one. Selectize can display a friendly string to the user, and have a more rich object value behind the scenes that includes the source / type. I'd prefer to take the burden of parsing off the server and let the UI manage how it displays these things (which can be in a single search) separately from how it's sent to the server.